Plugins/Bitbucket/Settings.cs

```diff
@@ -10,9 +10,9 @@ namespace Bitbucket
     public class Settings
     {
         private const string BitbucketHttpRegex =
-            @"https?:\/\/([\w\.\:]+\@)?(?<url>([a-zA-Z0-9\.\-\/]+?)):?(\d+)?\/scm\/(?<project>~?([\w\-]+?))\/(?<repo>([\w\-]+)).git";
+            @"https?:\/\/([\w\.\:]+\@)?(?<url>([a-zA-Z0-9\.\-\/]+?)):?(\d+)?\/scm\/(?<project>~?([\w\-]+?))\/(?<repo>([\w\-\.]+)).git";
```

Comment:
Why not allow all but / instead?
```suggestion
            @"https?:\/\/([\w\.\:]+\@)?(?<url>([a-zA-Z0-9\.\-\/]+?)):?(\d+)?\/scm\/(?<project>~?([^\/]+?))\/(?<repo>([^\/]+)).git";
```

Reply:
I do agree that for parsing purpose **repo** pattern, as well as **url** and **project** patterns, can be simplified to `[^\/]`. But I just don't dare to reduce validation potential of that pattern, in case somebody had plans to use it elsewhere. `\w\-\.` should cover all the valid characters of Bitbucket URL, so it should be fine as it is.

Reply:
https://docs.atlassian.com/bitbucket-server/javadoc/5.1.0/api/reference/com/atlassian/bitbucket/project/Project.html#KEY_REGEXP
Atlassian does not document properly it seems.

It is not a major problem to try use something non conforming as a BB server, it is just that w
error message is different
Of course similar for ssh

```suggestion
            @"https?:\/\/([\w\.\:]+\@)?(?<url>([a-zA-Z0-9\.\-\/]+?)):?(\d+)?\/scm\/(?<project>~?([^\/]+?))\/(?<repo>(.*?)).git";
```

Reply:
Your link is for **project key** pattern. I guess you meant to share [Project NAME_REGEXP](https://docs.atlassian.com/bitbucket-server/javadoc/5.1.0/api/reference/com/atlassian/bitbucket/project/Project.html#NAME_REGEXP) or [Repository NAME_REGEXP](https://docs.atlassian.com/bitbucket-server/javadoc/5.1.0/api/reference/com/atlassian/bitbucket/repository/Repository.html#NAME_REGEXP). Either way, suggested pattern will solve my issue as well. I will update PR shortly. Thank you!

Reply:
I couldn't help but notice that SSH pattern is different. Shouldn't it be exactly the same except for protocol name?